src/common/slash.ts

```diff
@@ -0,0 +1,27 @@
+import { ApplicationCommandOptionData, CommandInteraction } from 'discord.js';
+import { IContainer } from './types';
+import { z } from 'zod';
+
+export const slashCommands: Map<string, ISlashCommand> = new Map();
+
+export const SlashCommand = z.object({
+  commandName: z.string(),
+  name: z.string(),
+  description: z.string(),
+  parameters: z.array(z.any()),
+  execute: z.function(),
```

Comment:
We can make this a bit more specific in order to prevent the cast on line 89 in `src/common/bot.ts`
```suggestion
  execute: z
    .function()
    .args(z.any()) // Add more specific check later?
    .returns(z.union([z.void(), z.promise(z.void())])),
```